x-pack/plugins/endpoint/common/types.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export class EndpointAppConstants {
+  static ENDPOINT_INDEX_NAME = 'endpoint-agent*';
+}
+
+export interface EndpointResultList {
+  // the endpoint restricted by the page size
+  endpoints: EndpointData[];
+  // the total number of unique endpoints in the index
+  total: number;
+  // the page size requested
+  request_page_size: number;
+  // the index requested
+  request_page_index: number;
+}
+
+export interface EndpointData {
+  machine_id: string;
+  created_at: Date;
+  host: {
+    name: string;
+    hostname: string;
+    ip: string;
+    mac_address: string;
+    os: {
+      name: string;
+      full: string;
+    };
+  };
+  endpoint: {
+    domain: string;
+    is_base_image: boolean;
+    active_directory_distinguished_name: string;
+    active_directory_hostname: string;
+    upgrade: {
+      status?: string;
+      updated_at?: Date;
+    };
+    isolation: {
+      status: boolean;
+      request_status?: string | boolean;
+      updated_at?: Date;
+    };
+    policy: {
+      name: string;
+      id: string;
+    };
+    sensor: {
```

Comment:
We should change this to `endpoint`.  `sensor` is an SMP term and will not be used in the Elastic security product.